frappe/public/js/frappe/form/control.js

```diff
@@ -667,6 +667,46 @@ frappe.ui.form.ControlCurrency = frappe.ui.form.ControlFloat.extend({
 
 frappe.ui.form.ControlPercent = frappe.ui.form.ControlFloat;
 
+frappe.ui.form.ControlColor = frappe.ui.form.ControlData.extend({
+	make_input: function () {
+		this._super();
+		this.make_color_input();
+	},
+	make_color_input: function() {
+		var me = this
+		this.$input.addClass("color-input")
+		var color = ["#FFC4C4", "#ffe8cd", "#ffd2c2", "#ff8989", "#ffd19c", "#ffa685", "#ff4d4d", "#ffb868", "#ff7846", "#a83333", "#a87945", "#a84f2e", "#d2d2ff", "#f8d4f8", "#dac7ff", "#a3a3ff", "#f3aaf0", "#b592ff", "#7575ff", "#ec7dea", "#8e58ff", "#4d4da8", "#934f92", "#5e3aa8", "#ebf8cc", "#ffd7d7", "#d2f8ed", "#d9f399", "#ffb1b1", "#a4f3dd", "#c5ec63", "#ff8989", "#77ecca", "#7b933d", "#a85b5b", "#49937e", "#fffacd", "#d2f1ff", "#cef6d1", "#fff69c", "#a6e4ff", "#9deca2", "#fff168", "#78d6ff", "#6be273", "#a89f45", "#4f8ea8", "#428b46"];
+		console.log("parentS", ($(this).parents()));
+		this.$input.parents(".control-input-wrapper").append(`<div class="color-picker"><div class="color-picker-pallet"></div></div>`)
+			color.forEach(function(entry) {
+				$(".color-picker-pallet").append('<div class="color-box" data-color='+ entry + ' style="background-color:' + entry +'"></div>');
+			});
+		this.color_picker();
+
+	},
+	get_color_box: function (hex) {
+		return `<div data-color="${hex}" style="background-color: ${hex}"></div>`
+	},
+	color_picker: function () {
```

Comment:
`color_picker` is not a good function name, it should indicate what it does, should be `bind_events`